mvi-core/src/main/kotlin/ru/surfstudio/mvi/flow/BaseFlowMiddleware.kt

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright 2022 Surf LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package ru.surfstudio.mvi.flow
+
+import ru.surfstudio.mvi.core.event.Event
+import ru.surfstudio.mvi.core.event.MviLifecycleEvent
+import ru.surfstudio.mvi.lifecycle.MapperLifecycleEvent
+
+/**
+ * Base middleware with support of lifecycle events
+ *
+ * To receive events, you need to add event that implements [MviLifecycleEvent]
+ * and to call the [bindLifecycleEventfecycleEvent] for the entity implementing the MviView interface
```

Comment:
To receive events, you need to add event which implements [MviLifecycleEvent]
and to call [bindsLifecycleEvent] for [MviView] implementation